work17/index.js

```diff
@@ -0,0 +1,191 @@
+    const ul = document.createElement("ul");
+    const fragment = document.createDocumentFragment();
+
+    ul.classList = "slider_item"
+    slideDiv.classList = "slider_slide";
+
+    for (let i = 0; i < imagesDataArray.length; i++) {
+        const li = document.createElement("li");
+        const img = document.createElement("img");
+
+        img.src = imagesDataArray[i].src;
+        li.classList = "slider_items";
+        if (!i) {
+            li.classList.add("is-display");
+        }
+        li.id = `js-sliderItems${i + 1}`;
+
+        li.appendChild(img);
+        fragment.appendChild(li);
+    }
+
+    document.body.appendChild(wrapDiv)
+        .appendChild(slideDiv).appendChild(ul).appendChild(fragment);
+    slideDiv.appendChild(renderPagingNumber());
+    renderSlideArrow();
+    clickedSliderEvent();
+}
+
+const renderPagingNumber = () => {
+    let count = 1;
+    const sliderListElements = [...document.querySelectorAll(".slider_items")];
+
+    const span = document.createElement("span");
+    span.classList = "slider_pageNum";
+
+    sliderListElements.forEach(element => {
+        if (element.classList.contains("is-display")) {
+            span.textContent = `${count} / ${sliderListElements.length}`;
+            return span;
+        }
+    });
+    return span;
+};
+
+const renderSlideArrow = () => {
+    const wrapDiv = document.querySelector(".slider_slide");
+    const divLeft = document.createElement("div");
+    const divRight = document.createElement("div");
+    const buttonLeft = document.createElement("button");
+    const buttonRight = document.createElement("button");
+    const iLeft = document.createElement("i");
+    const iRight = document.createElement("i");
+
+    divLeft.classList = "slider_arrowBox-left slider_arrowBox";
+    divRight.classList = "slider_arrowBox-right slider_arrowBox";
+    buttonLeft.classList = "slider_prev slider_button"
+    buttonRight.classList = "slider_next slider_button"
+    buttonLeft.type = "button";
+    buttonLeft.disabled = true;
+    buttonRight.type = "button";
+    iLeft.classList = "fas fa-angle-left fa-3x";
+    iRight.classList = "fas fa-angle-right fa-3x";
+
+    divLeft.appendChild(buttonLeft).appendChild(iLeft);
+    divRight.appendChild(buttonRight).appendChild(iRight);
+
+    wrapDiv.appendChild(divLeft);
+    wrapDiv.appendChild(divRight);
+}
+
+const addOrRemoveDisabled = (index, nextOrPrevButton) => {
+    const sliderListElements = document.querySelector(".slider_item");
+    const targetElem = document.getElementById(`js-sliderItems${index}`);
+
+    const firstElem = sliderListElements.firstElementChild;
+    const lastElem = sliderListElements.lastElementChild;
+
+    if (targetElem === firstElem || targetElem === lastElem) {
+        nextOrPrevButton.disabled = true;
+    } else {
+        const disabledElem = document.querySelector("[disabled]");
+        disabledElem && disabledElem.removeAttribute("disabled");
+    }
```

Comment:
Q
Why do you get the element again when you have a conditional branch?
Since else is a process when it is not firstElem or lastElem,
You don't seem to need to get the element again.

Reply:
日本語で失礼します。

if文の中身では、li要素を一括で取得(sliderListElements)して、最初の要素(fristElem)と最後の要素(lastElem)が同じだった時にdisabledしています。
それ以外の時には、disabledを外すようにしたかったのですが、
else文で

> nextOrPrevButton.disabled = false;

としてしまうと、次へボタン(右矢印)で2枚目にいくと、nextOrPrevButtonの値は、

> <button class="slider_next slider_button" type="button">

が入っているため、戻るボタン(左矢印)のdisalbedが外れませんでした。

なので、結局は、button要素にdisabledがついている要素を取得して、disabledをremoveAttributeすれば、if文が冗長になることなく、すっきり書けると考えたので、今回のようなコードにしました。

回答になっているでしょうか？

Reply:
ご回答ありがとうございます。
現状は、clickedSliderEventの中でループ処理したnextOrPrevButtonを渡していると思うのですが、
ここでは、slider_prevとslider_nextを取得するコードを追加するのは、どうでしょうか？
コードがよりシンプルになるのではないかと思いました。
やすさんの意図していることと違ったらすみません。

またこれはnitsですが、
引数のindexに呼び出し元で代入される値は、1から始まるcurrentImgCountのため、
ここでの仮引数もcurrentImgCountの方がわかりやすいのではないか、と思いました。

```JavaScript

const addOrRemoveDisabled = (currentImgCount) => {
  const sliderListElements = document.querySelector(".slider_list");
  const targetElem = document.getElementById(`js-sliderItems${currentImgCount}`);

  const firstElem = sliderListElements.firstElementChild;
  const lastElem = sliderListElements.lastElementChild;

  const nextButton = document.querySelector(".slider_next");
  const prevButton = document.querySelector(".slider_prev");

  prevButton.disabled = targetElem === firstElem;
  nextButton.disabled = targetElem === lastElem;
};
```